### DISCUSSION TITLE
URI Handler

### DISCUSSION THREAD

**Discussion by Gameniti** [1Remote/1Remote#404] [Ideas]
**Is your feature request related to a problem? Please describe.**
Not related to a problem.

**Describe the solution you'd like**
I would find it really useful if a URI protocol handler would be added so that you can open a session via a web browser.

**Describe alternatives you've considered**

**Additional context**
Example:
in your web browser type `rdp://<ip>` and let PRemoteM automatically open a rdp connection where you can log in (or even better find a saved connection in PRemoteM and use credentials and option from there)

Something similar would be ssh where you type `ssh://<user>@<ip>` and PRemoteM opens a session as this user (and credentials if already saved)

**Comment by VShawn** [comment]
I don't know if this feature is needed, and how many users need this feature...

And there's another approach for quick connect in https://github.com/1Remote/PRemoteM/issues/273 by using our launcher.

I have made some effort on it like:

![image](https://user-images.githubusercontent.com/10143738/197317254-db45360f-2f65-4d7a-ae0a-942d786baadb.png)

**Comment by vampix** [comment]
Hi this feature would be ideal to use with some Remote Connection tools we have. It opens an Reverse tunnel and then triggers "C:\Windows\sysnative\mstsc.exe /v:localhost:port" or "rpd://localhost:port" to have the 1Remote to open this in tabs would help.

**Comment by majkinetor** [comment]
RDP needs username and pass to. So `rdp://<ip>` won't work unless ip is recognized in 1RM database which seems like defating the purpose... So you will need `rdp://<user>:<password>@<ip-or-name>` or similar and that looks problematic as browser can cache such URL with password.
